randovania/resolver/resolver.py

```diff
@@ -81,11 +81,19 @@ def _should_check_if_action_is_safe(state: State,
     return False
 
 
+class ResolverTimeout(Exception):
+    pass
+
+def _check_attempts(max_attempts: Optional[int]):
+    if max_attempts is not None and debug.get_attempts() >= max_attempts:
```

Comment:
I don't really like that part of the functionality depends on functionality of the `debug` module.

Reply:
I can pull this out; as far as I can tell this functionality isn't even used anywhere else besides commented out code lol

Reply:
Isn't this the only code that raises `ResolverTimeout`, which you explicitly look for?

Reply:
no I mean `debug`'s attempts is only used here otherwise:
```py
def increment_attempts():
    global count
    count += 1
    # if count > 1500:
    #     raise SystemExit
```
so i can pretty safely track the attempts elsewhere lol

Reply:
I had 100% forgot there was already a counter there for that random debugging lol